scripts/ruff.sh

```diff
@@ -0,0 +1,53 @@
+#!/usr/bin/env bash
+# SPDX-License-Identifier: Apache-2.0
+set -e
+
+# wrapper to combine ruff check, ruff format, and isort
+#
+# "ruff.sh fix" runs fixes and reformats the code
+# "ruff.sh check" checks style, format, and isort
+# "ruff.sh <args>" passes abitrary args to ruff
+
+if [ -z "$1" ]; then
+    echo "USAGE: $0 [check|fix|<args>]" >&2
+    exit 2
+fi
+
+run() {
+    declare -i err
+
+    echo "RUN: '$*'"
+    "$@"
+    err=$?
+    echo
+    return $err
+}
+
+case $1 in
+    "check")
+        declare -i exitcode=0
+
+        set +e
+        run ruff check .
+        exitcode=$(( exitcode + $? ))
+
+        run ruff format --diff .
+        exitcode=$(( exitcode + $? ))
+
+        run isort --check --diff .
+        exitcode=$(( exitcode + $? ))
+        set -e
+
+        if [ $exitcode -ne 0 ]; then
+            echo "ERROR: one or more checks have failed" >&2
+            exit 3
+        fi
```

Comment:
Can we help the user a bit more? How about having an array with the failed checks and printing it with the error message? So the user knows which check failed.

Reply:
ruff and isort are already printing detailed error information to stderr. Here is an example with a missing isort header and an extra newline:

```
$ tox -e ruff -- check
ruff: install_deps> python -I -m pip install isort==5.11.5 ruff==0.3.4
ruff: commands[0]> ./scripts/ruff.sh check
RUN: 'ruff check .'
All checks passed!

RUN: 'ruff format --diff .'
--- src/instructlab/lab.py
+++ src/instructlab/lab.py
@@ -28,7 +28,6 @@
     import torch
 
 
-
 class Lab:
     """Lab object holds high-level information about ilab CLI"""
 

1 file would be reformatted, 46 files already formatted

RUN: 'isort --check --diff .'
ERROR: src/instructlab/lab.py Imports are incorrectly sorted and/or formatted.
--- src/instructlab/lab.py:before        2024-04-22 09:42:27.262042
+++ src/instructlab/lab.py:after 2024-04-22 09:42:39.052428
@@ -12,6 +12,7 @@
 import sys
 import typing
 
+# Third Party
 from click_didyoumean import DYMGroup
 from git import GitError, Repo
 from huggingface_hub import hf_hub_download
Skipped 5 files

ERROR: one or more checks have failed
ruff: exit 3 (0.31 seconds) /home/heimes/redhat/ai/instructlab/instructlab> ./scripts/ruff.sh check pid=384394
  ruff: FAIL code 3 (4.46=setup[4.16]+cmd[0.31] seconds)
  evaluation failed :( (4.98 seconds)
```

Reply:
I could add `Run 'tox -e ruff' to auto-correct all fixable errors` to the output. WDYT?

Reply:
The error message now points to `tox -e ruff`